contracts/src/LightClient.sol

```diff
@@ -418,36 +498,45 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         return blockNumber - prevBlock > threshold;
     }
 
-    /// @notice get the number of L1 block updates
-    function getStateUpdateBlockNumbersCount() public view returns (uint256) {
-        return stateUpdateBlockNumbers.length;
-    }
-
     /// @notice get the HotShot commitment that represents the Merkle root containing the leaf at
-    /// the provided height
-    /// @param hotShotBlockHeight hotShotBlockHeight
+    /// the provided HotShot height
+    /// @param hotShotBlockHeight the HotShot block height
+    /// @return HotShotCommitment the HotShot commitment
     function getHotShotCommitment(uint256 hotShotBlockHeight)
         public
         view
         returns (HotShotCommitment memory)
     {
-        uint256 commitmentsHeight = hotShotCommitments.length;
-        if (hotShotBlockHeight >= hotShotCommitments[commitmentsHeight - 1].blockHeight) {
+        uint256 commitmentsHeight = stateHistoryCommitments.length;
+        if (
+            hotShotBlockHeight
+                >= stateHistoryCommitments[commitmentsHeight - 1].hotShotCommitment.blockHeight
+        ) {
             revert InvalidHotShotBlockForCommitmentCheck();
         }
-        for (uint256 i = 0; i < commitmentsHeight; i++) {
+        for (uint256 i = stateHistoryFirstIndex; i < commitmentsHeight; i++) {
             // The first commitment greater than the provided height is the root of the tree
             // that leaf at that HotShot height
-            if (hotShotCommitments[i].blockHeight > hotShotBlockHeight) {
-                return hotShotCommitments[i];
+            if (stateHistoryCommitments[i].hotShotCommitment.blockHeight > hotShotBlockHeight) {
+                return stateHistoryCommitments[i].hotShotCommitment;
             }
         }
 
-        return hotShotCommitments[commitmentsHeight - 1];
+        return stateHistoryCommitments[commitmentsHeight - 1].hotShotCommitment;
+    }
+
+    /// @notice get the number of state history commitments
+    /// @return uint256 The number of state history commitments
+    function getStateHistoryCount() public view returns (uint256) {
+        return stateHistoryCommitments.length;
     }
 
-    /// @notice get the number of HotShot block commitments
-    function getHotShotBlockCommitmentsCount() public view returns (uint256) {
-        return hotShotCommitments.length;
+    /// @notice set Max Block States allowed
+    /// @param historySeconds The maximum duration worth of state history updates to store based on
+    /// the block timestamp
+    function setMaxStateHistoryDuration(uint32 historySeconds) public onlyOwner {
+        if (historySeconds < 86400) revert InvalidMaxStateHistory();
```

Comment:
Is this meant to throw if the new value is smaller than the current value instead?

Reply:
That's another criterion to consider, which I responded to in the comment here (https://github.com/EspressoSystems/espresso-sequencer/pull/1670#discussion_r1716889601). I also believe there should be a minimum state history of one day—unless you think a shorter duration would be more appropriate?

Reply:
done here https://github.com/EspressoSystems/espresso-sequencer/commit/703546faeeee4ea9614ac3057053fd3ec8852767